base/applications/taskmgr/procpage.c

```diff
@@ -1240,14 +1241,18 @@ void ProcessPage_OnOpenFileLocation(void)
         goto Cleanup;
 
     /* Build the shell command line */
-    pszCmdLine = HeapAlloc(GetProcessHeap(), 0, (dwLength + CONST_STR_LEN(L"/select,\"\"")) * sizeof(WCHAR));
+    dwLength += CONST_STR_LEN(szCmdFormat) - CONST_STR_LEN(L"%s");
+    pszCmdLine = HeapAlloc(GetProcessHeap(), 0, dwLength * sizeof(WCHAR));
     if (!pszCmdLine)
         goto Cleanup;
 
-    StringCchPrintfW(pszCmdLine, dwLength + CONST_STR_LEN(L"/select,\"\""), L"/select,\"%s\"", pszExePath);
+    StringCchPrintfW(pszCmdLine, dwLength, szCmdFormat, pszExePath);
 
-    /* Call the shell to open the file location and select it */
-    ShellExecuteW(NULL, L"open", L"explorer.exe", pszCmdLine, NULL, SW_SHOWNORMAL);
+    /* Call the shell to open the file location and select it. If explorer shell
+     * is not available, use ReactOS's alternative file browser instead. */
```

Comment:
```suggestion
     * is not available, use ReactOS's alternative file browser instead. */
    /* FIXME: Use "explorer.exe /separate" instead of "filebrowser.exe" when implemented. */
```

Reply:
Now I think filebrowser.exe may not considered a workaround.
It's advantages over explorer.exe with `/separate` command are:
- Smaller binary size & simpler startup code
- Can be used along with taskmgr as a "minimal" shell
- Windows XP/2003-compatible, easy to use along with ReactOS taskmgr on Windows (just put it in the same folder)

Reply:
It's not very Windows XP/2003 compatible if you have to copy more ROS files over. It's very much ROS specific.
I don't disagree with you that this is a best solution for now, but `filebrowser.exe` may be removed in the future. This is very much a workaround for the missing `/separate` functionality.

Reply:
> In fact, from my testing Windows 7 doesn't even use /separate when opening the file location of a process

Windows 7 uses `SHOpenFolderAndSelectItems` but we uses the simpler way (`ShellExecute` explorer.exe with `/select,<Exe Path>` option) this gave the same results and also avoid introducing additional ole32 dependency.

Reply:
I'd argue what we do gives better results since it still works if the shell isn't loaded unlike Windows 7.

Still, I'd rather implement this using `/separate` when we do have this functionality. On Windows 7 using this option allows you to load the file browser without loading the shell and our implementation should do the same.

Reply:
I've found the solution:
`explorer.exe /separate,/select,"<Exe Path>"`

Source docs:
[https://superuser.com/questions/21394/explorer-command-line-switches](https://superuser.com/questions/21394/explorer-command-line-switches)
[https://web.archive.org/web/20150602220856/https://support.microsoft.com/en-us/kb/152457](https://web.archive.org/web/20150602220856/https://support.microsoft.com/en-us/kb/152457)

It works on ReactOS and also on Win10. No workaround needed anymore.

Reply:
I checked Windows 7 and you're right; you can't reliably use select and separate together. Therefore, this is a ROS-specific enhancement and should be labeled as such.